### DISCUSSION TITLE
How to I merge two action.yml?

### DISCUSSION THREAD

**Discussion by bencheng1990** [community/community#25528] [Actions]
<p>I would like my repo to have 2 functions:</p>
<ol>
<li>Create a release on tag push</li>
<li>Test my software in a docker environment</li>
</ol>
<p>Both require an action.yml in the repo. How do I combine them?</p>
<pre><code>name: "Upload a Release Asset"
description: "Upload a release asset to an existing release on your repository"
author: "Github"
inputs:
  upload_url:
    description: "The URL for uploading assets to the release"
    required: true
  asset_path:
    description: "The path to the asset you want to upload"
    required: true
  asset_name:
    description: "The name of the asset you want to upload"
    required: true
  asset_content_type:
    description: "The content-type of the asset you want to upload. See the supported Media Types here: https://www.iana.org/assignments/media-types/media-types.xhtml for more information"
    required: true
outputs:
  browser_download_url:
    description: "The URL users can navigate to in order to download the uploaded asset"
runs:
  using: "node12"
  main: "dist/index.js"
branding:
  icon: "package"
  color: "gray-dark"

name: 'Hello World'
description: 'Greet someone and record the time'
inputs:
  who-to-greet: # id of input
    description: 'Who to greet'
    required: true
    default: 'World'
outputs:
  time: # id of output
    description: 'The time we greeted you'
runs:
  using: 'docker'
  image: 'Dockerfile'
  args:
    - ${{ inputs.who-to-greet }}
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>It’s not supported to merge javascript and docker action in one metadata file.  If you’d like your repo have 2 functions, you can put your two actions into different directories, eg: .github/actions/action-a and .github/actions/action-b, it’s mentioned <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-actions#choosing-a-location-for-your-action" rel="nofollow noopener">here</a>, hope it helps.</p>
